src/http-api/errors.ts

```diff
@@ -63,11 +68,32 @@ export class MatrixError extends HTTPError {
         if (url) {
             message = `${message} (${url})`;
         }
-        super(`MatrixError: ${message}`, httpStatus);
+        super(`MatrixError: ${message}`, httpStatus, httpHeaders);
         this.errcode = errorJson.errcode;
         this.name = errorJson.errcode || "Unknown error code";
         this.data = errorJson;
     }
+
+    /**
+     * @returns the recommended delay in milliseconds to wait before retrying
+     * the request that triggered this error, or null if no delay is recommended.
```

Comment:
it feels like this method should distinguish between "This error doesn't look like it should have a retry-after value" (ie, it's not a 429 nor an M_LIMIT_EXCEEDED) and "This error looks like it should have a retry-after value, but it doesn't have one)

Reply:
Good idea, there can be another method for that.  The reason this one doesn't check the errcode is because callers tend to have an if-else chain on the errcode before checking `retry_after_ms`, so putting the check here too would be redundant.

Reply:
b31179a67